### DISCUSSION TITLE
Performance Benchmark for jmeter-iso8583

### DISCUSSION THREAD

**Discussion by mehrdad2000** [tilln/jmeter-iso8583#89] [General]
Hi
Is there any Performance Benchmark availabe for jmeter-iso8583?
what is the maximum throughput that get with jmeter-iso8583? 1K TPS? 2K TPS? scenario details? environment condition?

Any idea?
Thanks

**Comment by tilln** [comment]
> Is there any Performance Benchmark availabe for jmeter-iso8583?

No, there is not.
This would depend not only on the capacity of your VM running JMeter but also significantly on the message payload, in particular the number of cryptographic operations for each message.

Your best option is to benchmark your script and VM yourself by running against a stub (e.g. via Response Listener) instead of the real system.

**Reply by mehrdad2000** [reply]
@tilln Ok, I have performance issue, seems generate some message take time, what is the solution here??
Is it possible to run multiple jmeter instance in single host? 
Or I should use multi node jmeter on multiple host?

Any trick or tunning tips?
Thanks

**Reply by tilln** [reply]
If you are using your custom JMSChannel I would recommend profiling and tuning that code first before doing anything else.

**Reply by mehrdad2000** [reply]
@tilln what about ascci channel? What should i do if have same issue when i use default channels?

**Reply by tilln** [reply]
You still want to find out what takes time and if that can be improved.
You could try to insert logging statements in your script, e.g. in Pre/Post-Processors around the components/samplers that appear slow.
Debug logging may also help so you can parse the log output for timestamps.

Can you describe your performance problem a bit more and perhaps explain what your script is trying to do?

**Reply by mehrdad2000** [reply]
@tilln 1-I try to enable debug and profiling and logging statment pre/post and still can’t find any clue, but i’m still keep trying.
2-For simple message like echo 0800 What is your highest TPS you get?

**Reply by tilln** [reply]
> 2-For simple message like echo 0800 What is your highest TPS you get?

Not sure how this would help you.
I'm guessing around 500-1000.

**Reply by mehrdad2000** [reply]
@tilln did you try more than 1000 number? Did you remember due to botelneck or limitation set 1000 or you can exceed this number?

**Reply by tilln** [reply]
No, I haven't tried, so I have no idea where the bottleneck may be, sorry.
How much are you aiming for?
You may just have to scale out to multiple JMeter instances.
